build-logic/uber-plugins/src/main/kotlin/gradlebuild.kotlin-library.gradle.kts

```diff
@@ -65,16 +65,20 @@ tasks {
 fun KotlinCompile.configureKotlinCompilerForGradleBuild() {
     kotlinOptions {
         incremental = true
-        allWarningsAsErrors = true
+        /*
+          w: Flag is not supported by this version of the compiler: -Xskip-runtime-version-check
+          w: Language version 1.4 is deprecated and its support will be removed in a future version of Kotlin
+          e: warnings found and -Werror specified
+         */
+        // allWarningsAsErrors = true
```

Comment:
I think it'd make sense to split the configuration between this plugin and the  `gradlebuild.internal.kotlin` plugin so we could remove the `languageVersion` setting and re-enable `allWarningsAsErrors` for the internal ones such as the configuration-cache modules.

That can be done in a subsequent PR.

Reply:
+1 to do this because this PR alone opens up for more warnings to get unnoticed.